client/api/omni/specs/compression.go

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package specs
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/klauspost/compress/zstd"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	defaultCompress = true
+)
+
+var jsonMarshaler = protojson.MarshalOptions{
+	UseProtoNames:  true,
+	UseEnumNumbers: true,
+}
+
+func doCompress(r io.Reader) ([]byte, error) {
+	var buf bytes.Buffer
+
+	writer, err := zstd.NewWriter(&buf)
```

Comment:
I believe there was a way to re-use the `ztsd` instance to re-use buffers on compression/decompression

I think it'd cool to have a pool of slices one way or another, but it'd be tricky to track the lifecycle of these...

Reply:
addressed both